Testing - Backend - The TestJobApi/TestJobApis test is not stable

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/4415/kubeflow-pipeline-e2e-test/1298513122180468740
```
integration-test-k95mb-63616182: === RUN   TestJobApi/TestJobApis
integration-test-k95mb-63616182: --- FAIL: TestJobApi (83.28s)
integration-test-k95mb-63616182:     --- FAIL: TestJobApi/TestJobApis (41.45s)
integration-test-k95mb-63616182:         job_api_test.go:177: 
integration-test-k95mb-63616182:             	Error Trace:	job_api_test.go:177
integration-test-k95mb-63616182:             	Error:      	Not equal: 
integration-test-k95mb-63616182:             	            	expected: "hello world"
integration-test-k95mb-63616182:             	            	actual  : "argument parameter"
integration-test-k95mb-63616182:             	            	
integration-test-k95mb-63616182:             	            	Diff:
integration-test-k95mb-63616182:             	            	--- Expected
integration-test-k95mb-63616182:             	            	+++ Actual
integration-test-k95mb-63616182:             	            	@@ -1 +1 @@
integration-test-k95mb-63616182:             	            	-hello world
integration-test-k95mb-63616182:             	            	+argument parameter
integration-test-k95mb-63616182:             	Test:       	TestJobApi/TestJobApis
integration-test-k95mb-63616182:         job_api_test.go:183: 
integration-test-k95mb-63616182:             	Error Trace:	job_api_test.go:183
integration-test-k95mb-63616182:             	Error:      	Not equal: 
integration-test-k95mb-63616182:             	            	expected: "argument parameter"
integration-test-k95mb-63616182:             	            	actual  : "hello world"
integration-test-k95mb-63616182:             	            	
integration-test-k95mb-63616182:             	            	Diff:
integration-test-k95mb-63616182:             	            	--- Expected
integration-test-k95mb-63616182:             	            	+++ Actual
integration-test-k95mb-63616182:             	            	@@ -1 +1 @@
integration-test-k95mb-63616182:             	            	-argument parameter
integration-test-k95mb-63616182:             	            	+hello world
integration-test-k95mb-63616182:             	Test:       	TestJobApi/TestJobApis
```

--- a/backend/test/integration/job_api_test.go
+++ b/backend/test/integration/job_api_test.go
@@ -137,6 +137,9 @@ func (s *JobApiTestSuite) TestJobApis() {
 	assert.Nil(t, err)
 
 	/* ---------- Create a new argument parameter job by uploading workflow manifest ---------- */
+	// Make sure the job is created at least 1 second later than the first one,
+	// because sort by created_at has precision of 1 second.
+	time.Sleep(1 * time.Second)
 	argParamsBytes, err := ioutil.ReadFile("../resources/arguments-parameters.yaml")
 	assert.Nil(t, err)
 	argParamsBytes, err = yaml.ToJSON(argParamsBytes)

--- a/backend/test/integration/upgrade_test.go
+++ b/backend/test/integration/upgrade_test.go
@@ -108,15 +108,16 @@ func (s *UpgradeTests) SetupSuite() {
 
 func (s *UpgradeTests) TearDownSuite() {
 	if *runIntegrationTests {
-		t := s.T()
-
-		// Clean up after the suite to unblock other tests. (Not needed for upgrade
-		// tests because it needs changes in prepare tests to persist and verified
-		// later.)
-		test.DeleteAllExperiments(s.experimentClient, t)
-		test.DeleteAllPipelines(s.pipelineClient, t)
-		test.DeleteAllRuns(s.runClient, t)
-		test.DeleteAllJobs(s.jobClient, t)
+		if !*isDevMode {
+			t := s.T()
+			// Clean up after the suite to unblock other tests. (Not needed for upgrade
+			// tests because it needs changes in prepare tests to persist and verified
+			// later.)
+			test.DeleteAllExperiments(s.experimentClient, t)
+			test.DeleteAllPipelines(s.pipelineClient, t)
+			test.DeleteAllRuns(s.runClient, t)
+			test.DeleteAllJobs(s.jobClient, t)
+		}
 	}
 }
 

